backend/packages/Upgrade/src/api/services/ExperimentAssignmentService.ts

```diff
@@ -1559,11 +1563,43 @@ export class ExperimentAssignmentService {
     return;
   }
 
+  private async getNewExperimentConditionAssignment(
+    experiment: Experiment,
+    user: ExperimentUser,
+    logger: UpgradeLogger,
+    enrollmentCount?: { conditionId: string; userCount: number }[]
+  ): Promise<ExperimentCondition> {
+    const isMoocletExperiment = this.moocletExperimentService.isMoocletExperiment(experiment.assignmentAlgorithm);
+
+    if (isMoocletExperiment && !env.mooclets.enabled) {
+      logger.error({
+        message: 'Mooclet experiment algorithm is indicated but mooclets are not enabled',
+        experiment,
+        user,
+      });
+      return undefined;
+    }
+
+    if (isMoocletExperiment && env.mooclets.enabled) {
+      return this.getConditionFromMoocletProxy(experiment, user);
+    } else {
+      return this.assignRandom(experiment, user, enrollmentCount);
+    }
+  }
+
+  private async getConditionFromMoocletProxy(experiment: Experiment, user: ExperimentUser) {
+    const userId = user.id;
+
+    const condition = await this.moocletExperimentService.getConditionFromMoocletProxy(experiment, userId);
+
+    return condition;
+  }
+
   private assignRandom(
     experiment: Experiment,
     user: ExperimentUser,
     enrollmentCount?: { conditionId: string; userCount: number }[]
-  ): ExperimentCondition {
```

Comment:
Is there a way to keep the return type here?

Reply:
yes probably, i can add that back, not sure why removed